subsys/usb/class/cdc_acm.c

```diff
@@ -422,9 +420,11 @@ static int cdc_acm_fifo_fill(struct device *dev,
 	len = len > sizeof(u32_t) ? sizeof(u32_t) : len;
 #endif
 
-	usb_write(CONFIG_CDC_ACM_IN_EP_ADDR, tx_data, len, &bytes_written);
-
-	return bytes_written;
+	err = usb_write(CONFIG_CDC_ACM_IN_EP_ADDR, tx_data, len, &wrote);
+	if (err != 0) {
+		return err;
```

Comment:
This seems to contradict the docstring `@return Number of bytes sent.`. What is the return value for this function call?

Reply:
I think this should be `if (err < 0)`, since return values are typically -ERRCODE on failure

Reply:
Hi @IvanSanchez @carlescufi stepping back, there's two types of return value in Zephyr:

- errors encoded as an int, where 0 means success
- result with an optional error.  Encoded as an int where >= 0 is the result and < 0 is the error.

If usb_write() succeeds then it sets wrote and returns 0.  If it fails, it writes nothing and returns an error code.  So testing against 0 is fine.

There's no effective difference between err < 0 and err != 0 it's just the latter is slightly more correct :)

Reply:
Agreed, there is no fundamental difference here